### DISCUSSION TITLE
How to set up a map with zones

### DISCUSSION THREAD

**Discussion by ghost** [Tasshack/dreame-vacuum#723] [Q&A] [answered]
Hello

I use this: 

`action: dreame_vacuum.vacuum_set_restricted_zone`
`data:`
`  zones:`
`    - - -444`
`      - -1986`
`      - 656`
`      - -1213`
`target:`
`  device_id: vacuum.vacuum`

after performing the action, the zone is created on the xiaomi map card but after refreshing the map, everything is deleted

**Comment by Tasshack** [comment]
Can you try latest Beta version?

**Comment by ghost** [comment]
My ver 2.0.0b16. with dreamehome support acount.

**Comment by ghost** [comment]
New initialization for vacuum cleaner integration fixes this, but after some time it's the same again. Probably disconnects me from the cloud. I don't know why? how to check it? I don't see it in the entities.

**Reply by Tasshack** [reply]
Vacuum does not respond if you try to send a command it does not understand and that causes the integration think that it has been disconnected and all entites becomes unavailable.

**Reply by ghost** [reply]
yes it's true but it's something different. Yesterday I could edit maps, but today I can't. I didn't use integration during this time. maybe it's the cloud's fault? If I can't do anything about it, how can I re-initiate the integration from the script level? thank you for your reply.

**Reply by Tasshack** [reply, marked as answer]
Hard reset the device.
